homeassistant/components/sensibo/quality_scale.yaml

```diff
@@ -0,0 +1,77 @@
+rules:
```

Comment:
```py
    async def async_set_temperature(self, **kwargs: Any) -> None:
        """Set new target temperature."""
        if "targetTemperature" not in self.device_data.active_features:
            raise HomeAssistantError(
                translation_domain=DOMAIN,
                translation_key="no_target_temperature_in_features",
            )
```
Isn't this guardable by the supported_features?

Reply:
idem
```py
    async def async_set_fan_mode(self, fan_mode: str) -> None:
        """Set new target fan mode."""
        if "fanLevel" not in self.device_data.active_features:
            raise HomeAssistantError(
                translation_domain=DOMAIN,
                translation_key="no_fan_level_in_features",
            )
```

Reply:
supported features are not continuously evaluated currently, we need to do that and then these could probably be removed.

Reply:
You mean we don't evaluate them in sensibo? Because in HA I believe we do

Reply:
The current implementation only checks once at init. So if it's allowed then it's always allowed (from HA's perspective) and therefore these guards as the api would not allow it.
Anyway, changing this so it's continuously evaluated

Reply:
Prel PR: https://github.com/home-assistant/core/pull/134412